rclpy/src/rclpy/_rclpy.c

```diff
@@ -4523,21 +4523,28 @@ rclpy_get_node_parameters(PyObject * Py_UNUSED(self), PyObject * args)
     return NULL;
   }
 
-  if (!PyDict_Contains(params_by_node_name, py_node_name_with_namespace)) {
-    // No parameters for current node.
-    Py_DECREF(params_by_node_name);
-    Py_DECREF(py_node_name_with_namespace);
-    return PyDict_New();
-  }
-  PyObject * node_params = PyDict_GetItem(params_by_node_name, py_node_name_with_namespace);
-  Py_DECREF(py_node_name_with_namespace);
-  if (NULL == node_params) {
-    Py_DECREF(params_by_node_name);
-    return NULL;
+  PyObject * node_params = PyDict_New();
```

Comment:
If `PyDict_New()` fails it will return NULL, which should be guarded against.

Reply:
Fixed. Good catch.